codes/c/chapter_divide_and_conquer/hanota.c

```diff
@@ -1,82 +1,75 @@
 /**
  * File: hanota.c
  * Created Time: 2023-10-01
- * Author: Zuoxun (845242523@qq.com)
+ * Author: Zuoxun (845242523@qq.com), lucas(superrat6@gmail.com)
  */
 
 #include "../utils/common.h"
 
+// 假设最多有 1000 个排列
+#define MAX_SIZE 1000
+
 /* 移动一个圆盘 */
-void move(vector *src, vector *tar) {
+void move(int *src, int *srcSize, int *tar, int *tarSize) {
     // 从 src 顶部拿出一个圆盘
-    int *panTemp = vectorBack(src);
-    int *pan = malloc(sizeof(int));
-    *pan = *panTemp;
-    vectorPopback(src);
+    int pan = src[*srcSize - 1];
+    src[*srcSize - 1] = 0;
+    (*srcSize)--;
     // 将圆盘放入 tar 顶部
-    vectorPushback(tar, pan, sizeof(int));
+    tar[*tarSize] = pan;
+    (*tarSize)++;
 }
 
 /* 求解汉诺塔：问题 f(i) */
-void dfs(int i, vector *src, vector *buf, vector *tar) {
+void dfs(int i, int *src, int *srcSize, int *buf, 
+            int *bufSize, int *tar, int *tarSize) {
     // 若 src 只剩下一个圆盘，则直接将其移到 tar
     if (i == 1) {
-        move(src, tar);
+        move(src, srcSize, tar, tarSize);
         return;
     }
     // 子问题 f(i-1) ：将 src 顶部 i-1 个圆盘借助 tar 移到 buf
-    dfs(i - 1, src, tar, buf);
+    dfs(i - 1, src, srcSize, tar, tarSize, buf, bufSize);
     // 子问题 f(1) ：将 src 剩余一个圆盘移到 tar
-    move(src, tar);
+    move(src, srcSize, tar, tarSize);
     // 子问题 f(i-1) ：将 buf 顶部 i-1 个圆盘借助 src 移到 tar
-    dfs(i - 1, buf, src, tar);
+    dfs(i - 1, buf, bufSize, src, srcSize, tar, tarSize);
 }
 
 /* 求解汉诺塔 */
-void solveHanota(vector *A, vector *B, vector *C) {
-    int n = A->size;
+void solveHanota(int *A, int *ASize, int *B, int *BSize, int *C, int *CSize, int n) {
```

Comment:
There is an extra parameter `int n`.
I don't know how to change it directly on top of your commit, please @krahets  help!